sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -3723,3 +3723,40 @@ case class Luhncheck(input: Expression) extends RuntimeReplaceable with Implicit
   override protected def withNewChildrenInternal(
       newChildren: IndexedSeq[Expression]): Expression = copy(newChildren(0))
 }
+
+/**
+ * A function that prepends a backslash to each instance of single quote
+ * in the given string and encloses the result by single quotes.
+ */
+@ExpressionDescription(
+  usage = """
+    _FUNC_(str) - Returns `str` enclosed by single quotes and
+    each instance of single quote in it is preceded by a backslash.
```

Comment:
It would be better to leave one string because the new line character occurs in docs.